File: webapp/src/main/vueapp/src/logic/models/Office.ts
import {GenericEntity} from '~/logic/models/utils/GenericEntity';
import {Locality} from '~/logic/models/Locality';

export class Office extends GenericEntity<Office> {
    private _id: number;
    private _name: string;
    private _phone: string;
    private _email: string;
    private _street: string;
    private _url: string;
    private _localityId: number;

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get name(): string {
        return this._name;
    }

    public set name(value: string) {
        this._name = value;
    }

    public get phone(): string {
        return this._phone;
    }

    public set phone(value: string) {
        this._phone = value;
    }

    public get email(): string {
        return this._email;
    }

    public set email(value: string) {
        this._email = value;
    }

    public get street(): string {
        return this._street;
    }

    public set street(value: string) {
        this._street = value;
    }

    public get url(): string {
        return this._url;
    }

    public set url(value: string) {
        this._url = value;
    }

    public get localityId(): number {
        return this._localityId;
    }

    public set localityId(value: number) {
        this._localityId = value;
    }
}

File: models/src/main/java/ar/edu/itba/paw/models/Office.java
package ar.edu.itba.paw.models;

import javax.persistence.*;

@Entity
@Table(
        name = "office",
        indexes = {
                @Index(columnList = "office_id", name = "office_office_id_uindex", unique = true),
        }
)
public class Office extends GenericModel<Integer> {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "office_office_id_seq")
    @SequenceGenerator(sequenceName = "office_office_id_seq", name = "office_office_id_seq", allocationSize = 1)
    @Column(name = "office_id")
    private Integer id;
    @Column(name = "phone")
    private String phone;
    @Column(name = "email")
    private String email;
    @Column(name = "name", nullable = false)
    private String name;
    @Column(name = "street", nullable = false)
    private String street;
    @Column(name = "url")
    private String url;
    @JoinColumn(name = "locality_id")
    @ManyToOne(fetch = FetchType.EAGER)
    private Locality locality;

    @Override
    public Integer getId() {
        return this.id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhone() {
        return this.phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getEmail() {
        return this.email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getStreet() {
        return this.street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public Locality getLocality() {
        return this.locality;
    }

    public void setLocality(Locality locality) {
        this.locality = locality;
    }

    public String getUrl() {
        return this.url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    @Override
    protected boolean isSameType(Object o) {
        return o instanceof Office;
    }
}
